src/haddock/libs/libcli.py

```diff
@@ -68,3 +68,39 @@ def add_output_dir_arg(ap):
         type=Path,
         default=Path.cwd(),
         )
+
+
+class _ParamsToDict(Action):
```

Comment:
This here apparently converts the arguments to a dictionary, but isn't it the same as what Argparse does?

https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args

Reply:
interesting point..this is actually the Class that we implemented for the haddock3-score CLI, just moved to the libcli library to make it accessible to other clis..
this is the argparse action that parses a parameter (`-p`) defined as a dictionary with all the haddock3 parameters (example haddock3-analyse -r run1-test -m 1 **-p dockq True irmsd True**)..I don't know if there's another argparse magic that can do it

Reply:
Sounds like one of the many "re-imagined" routines that has custom behaviour - best leave as it is since it might probably break a lot.